### ISSUE TITLE
paragraph styles are not applied within lists

### ISSUE THREAD

**Issue by Lumpusz** [plutext/docx4j-ImportXHTML#28]
Hi,

I have been trying to get css to style working for quite a while. All the help that I have found suggests that I am doing it right, still I don't get the expected output.
I have the following xhtml to convert to docx:
```html
<body
  <ol class="question">
    <li>
      <p class="question-text">Question</p>
      <ol class="question">
        <li>
          <p class="question-text">SubQuestion</p>
        </li>
      </ol>
    </li>
  </ol>
</body>
```

When I run it through XHTMLImporter the resulting xml is missing "question-text" styles:

```xml
<w:document ...>
    <w:body>
        <w:p>
            <w:pPr>
                <w:numPr>
                    <w:ilvl w:val="0"/>
                    <w:numId w:val="6"/>
                </w:numPr>
            </w:pPr>
            <w:r>
                <w:rPr>
                    <w:rFonts w:ascii="Times New Roman"/>
                </w:rPr>
                <w:t>Question</w:t>
            </w:r>
        </w:p>
        <w:p>
            <w:pPr>
                <w:numPr>
                    <w:ilvl w:val="1"/>
                    <w:numId w:val="6"/>
                </w:numPr>
            </w:pPr>
            <w:r>
                <w:rPr>
                    <w:rFonts w:ascii="Times New Roman"/>
                </w:rPr>
                <w:t>SubQuestion</w:t>
            </w:r>
        </w:p>
        ...
    </w:body>
</w:document>
```

I have a docx template with the 2 defined styles in it. The list style is applied, but the paragraph style is not.

```java
WordprocessingMLPackage wordMLPackage = WordprocessingMLPackage.load(FileMgr.getTemplate());
XHTMLImporterImpl XHTMLImporter = new XHTMLImporterImpl(wordMLPackage);
XHTMLImporter.setRunFormatting(FormattingOption.CLASS_TO_STYLE_ONLY);
XHTMLImporter.setParagraphFormatting(FormattingOption.CLASS_TO_STYLE_ONLY);
String unescaped = "<body><ol class=\"question\"><li><p class=\"question-text\">Question</p><ol class=\"question\"><li><p class=\"question-text\">SubQuestion</p></li></ol></li></ol></body>";
wordMLPackage.getMainDocumentPart().getContent().addAll(
     XHTMLImporter.convert(unescaped, ""));
wordMLPackage.save(outputFile);
```

This is what I would expect:
```xml
<w:document  ...>
    <w:body>
        <w:p>
            <w:pPr>
                <w:numPr>
                    <w:ilvl w:val="0"/>
                    <w:numId w:val="6"/>
                </w:numPr>
            </w:pPr>
            <w:r>
                <w:rPr>
                    <w:pStyle w:val="question-text"/>
                </w:rPr>
                <w:t>Question</w:t>
            </w:r>
        </w:p>
        <w:p>
            <w:pPr>
                <w:numPr>
                    <w:ilvl w:val="1"/>
                    <w:numId w:val="6"/>
                </w:numPr>
            </w:pPr>
            <w:r>
                <w:rPr>
                    <w:pStyle w:val="question-text"/>
                </w:rPr>
                <w:t>SubQuestion</w:t>
            </w:r>
        </w:p>
        ...
    </w:body>
</w:document>
```

What do I miss?

**Comment by plutext** [comment]
I guess you may still need CSS for eg "question-text" defined somewhere?  It shouldn't matter what format it specifies.

**Comment by Lumpusz** [comment]
Thanks for your reply.
IT is not that. I tried this:
```html
<html>
<head>
  <style>
    .question-text {
      display : block
    }
  </style>
</head>
<body>
  <ol class="question">
    <li>
      <p class="question-text">Question</p>
      <ol class="question">
        <li>
          <p class="question-text">SubQuestion</p>
        </li>
      </ol>
    </li>
  </ol>
</body>
```

"question-text' style seems to be ignored completely when generating the word xml.

**Comment by plutext** [comment]
A couple of other things to try/check:

1.  Try your p with class="question-text" outside the ol?  ie a sibling to it; child of body.

2. Unzip your docx template, inspect the styles part, and make sure the style id is "question-text" ie with the dash in it.

**Comment by Lumpusz** [comment]
1: It works as expected
```html
<html>
<head>
  <style>
    .question-text {
      display : block
    }
  </style>
</head>
<body>
  <p class=\"question-text\">MyParagraph</p>
  <ol class="question">
    ...
  </ol>
</body>
</html>
```
If &lt;p&gt; is outside of &lt;ol&gt;, style is applied:
```xml
...
<w:p>
  <w:pPr>
    <w:pStyle w:val="question-text"/>
  </w:pPr>
  <w:r>
    <w:rPr>
        <w:rFonts w:ascii="Times New Roman"/>
    </w:rPr>
    <w:t>MyParagraph</w:t>
  </w:r>
</w:p>
...
```
2: when I do this:
```java
wordMLPackage.getMainDocumentPart().getStyleTree().getParagraphStylesTree().toList().forEach(n -> System.out.println(n.data.getStyle().getName().getVal()));
```
I actually get back the style:
`DocDefaults`
`Normal`
`List Paragraph`
`question-text`

**Comment by Lumpusz** [comment]
This seem to be the case for tables as well.
```html
<html>
<head>
  <style>
    .somestyle {
      display : block
    }
  </style>
</head>
<body>
  <table>
    <tr><td><p class="somestyle">SomeText</p></td></tr>
  </table>
</body>
</html>
```
Results in:
```xml
<w:tc>
    <w:tcPr>
        <w:tcW w:w="292" w:type="dxa"/>
        <w:tcBorders>
            <w:top w:val="single" w:color="000000" w:sz="8"/>
            <w:left w:val="single" w:color="000000" w:sz="8"/>
            <w:bottom w:val="single" w:color="000000" w:sz="8"/>
            <w:right w:val="single" w:color="000000" w:sz="8"/>
        </w:tcBorders>
        <w:tcMar>
            <w:top w:w="15" w:type="dxa"/>
            <w:left w:w="15" w:type="dxa"/>
            <w:bottom w:w="15" w:type="dxa"/>
            <w:right w:w="15" w:type="dxa"/>
        </w:tcMar>
        <w:vAlign w:val="center"/>
    </w:tcPr>
    <w:p>
        <w:pPr/>
        <w:r>
            <w:rPr>
                <w:rFonts w:ascii="Times New Roman"/>
            </w:rPr>
            <w:t>SomeText</w:t>
        </w:r>
    </w:p>
</w:tc>
```

**Comment by Lumpusz** [comment]
OK, I think I found the answer. Here is the trick: the mentioned styles which should map to css class names are defined as "character" styles in the word tempalte. It seems that these styles are not applied to `<p/>`. One could workaround this by adding an explicit span tag inside `<p/>`. For example:
Instead of:
```html
...
  <p class="somestyle">SomeText</p>
...
```
```html
...
  <p><span class="somestyle">SomeText</span></p>
...
```
Solved the issue for both lists, and tables.
